addon/utils/dom-helpers.ts

```diff
@@ -494,3 +496,18 @@ export function getWindowSelection(): Selection {
 
   return selection;
 }
+
+export function getPathFromRoot(to: Node, inclusive: boolean): Node[] {
```

Comment:
is dom-helpers the right place for this method?

Reply:
> as with anything keeps references to dom elements, there's a risk for memory leakage and I didn't look into that. other than that seems to look and work fine.

I think its fine cause everything is implemented in immutable style, datastore gets rebuilt on every parse so old ones should be garbage collected